packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutThreshold.tsx

```diff
@@ -405,25 +406,34 @@ export const ChartDonutThreshold: React.FunctionComponent<ChartDonutThresholdPro
   donutHeight = Math.min(height, width),
   donutWidth = Math.min(height, width),
   innerRadius = (Math.min(donutHeight, donutWidth) - 34) / 2,
-
   ...rest
 }: ChartDonutThresholdProps) => {
+
   // Returns computed data representing pie chart slices
   const getComputedData = () => {
-    const computedData = [];
-    const datum = getData(data);
-    let prevYVal = 0;
-    datum.forEach((dataPoint: {_x?: any, _y: any}) => {
-      computedData.push({ x: dataPoint._x, y: dataPoint._y ? Math.abs(dataPoint._y - prevYVal) : 0 });
-      prevYVal = dataPoint._y;
-    });
-    computedData.push({ y: prevYVal ? Math.abs(100 - prevYVal) : 0 });
-    return computedData;
-  };
+    // Format and sort data. Sorting ensures thresholds are displayed in the correct order and simplifies calculations.
+    const datum = Data.formatData(data, {x, y, ...rest}, ['x', 'y']).sort((a: any,b: any) => a._y - b._y);
+
+    // Data must be offset so that the sum of all data point y-values (including the final slice) == 100.
+    const [prev, computedData] = datum.reduce((acc: [number, any], dataPoint: {_x: number | string, _y: number}) => {
+      return [
+        dataPoint._y, // Set the previous value to current y value
+        [
+          ...acc[1],
+          {
+            x: dataPoint._x, // Conditionally add x property only if it is in the original data object
+            y: dataPoint._y - acc[0] // Must be offset by previous value
+          }
+        ]
+      ];
+    }, [0, []]);
```

Comment:
Should we apply this to donut utilization as well?

Reply:
I almost did but thought it was probably outside of the scope of this PR.

Reply:
We could probably pull out the utilization/threshold data formatting function into a single util function. The logic is very similar, we just need to handle the difference between a single utilization data point and multiple threshold data points.

Reply:
Let's keep them separate. The code is complex and not easy to understand what's happening.